Update changelog for 1.2.3 (#1289)
astronomer/astro-sdk: python-sdk/docs/CHANGELOG.md

# Changelog

## 1.2.3

### Bug fixes
* When `if_exists` is set to `replace` in Dataframe operator, replace the table rather than append. This change fixes a regression on the Dataframe operator which caused it to append content to an output table instead of replacing.  [#1260](https://github.com/astronomer/astro-sdk/issues/1260)

> * Support `database` kwarg in `PostgresHook`. Since schema is renamed to database in airflow as per this [PR](https://github.com/apache/airflow/pull/26744), users can now set database instead of schema. [#1276](https://github.com/astronomer/astro-sdk/pull/1276)

Reviewer: I think we should add something similar to "Pass the table metadata database value to the underlying airflow Postgres hook correctly" also do we need the Airflow PR link here @phanikumv @sunank200 ?

Reviewer 2: Thats right- we arent doing any change to PostgresHook in this PR. Lets remove the airflow PR link and reword this

Author: But from user's perspective, the thing that would change is that we support `database` kwargs in `PoostgresHook` right?

Reviewer: earlier we were database kwargs value to `self.table.metadata.schema` which was not correct

Author: [Done](https://github.com/astronomer/astro-sdk/pull/1289/commits/166a8f7709641cb715f98b6a9bc54932a53192eb)

# Changelog

## 1.2.3

### Bug fixes

> * Fix the dataframe operator behavior of appending to it's output table instead of replacing. When `if_exists` is set to `replace`, replace the table rather than append [#1260](https://github.com/astronomer/astro-sdk/issues/1260)

Reviewer 2: ```suggestion
* When `if_exists` is set to `replace` in dataframe operator, replace the table rather than append. This bug fix fixes a regression on the Dataframe operator which caused it to append content to an output table instead of replacing.  [#1260](https://github.com/astronomer/astro-sdk/issues/1260)
```

Author: [Done](https://github.com/astronomer/astro-sdk/pull/1289/commits/7599ef157ac85a6992512c9367fb4368ce9f4e86)

> * Previously, `PostgresHook` did not support `database` kwarg. After schema is renamed to database in airflow as per this [PR](https://github.com/apache/airflow/pull/26744), now the user needs to set database to the table's database. [#1276](https://github.com/astronomer/astro-sdk/pull/1276)

Reviewer 2: ```suggestion
* Support `database` kwarg in `PostgresHook`. After schema is renamed to database in airflow as per this [PR](https://github.com/apache/airflow/pull/26744), the user can now set database instead of schema. [#1276](https://github.com/astronomer/astro-sdk/pull/1276)
```

Reviewer 3: https://astronomer.slack.com/archives/C03868KGF2Q/p1669131445686549?thread_ts=1669130287.404939&cid=C03868KGF2Q -- What about backwards compatibility with old postgres provider?

We need to think this through before releasing this change

Author: Created an issue to check backward compatibility for PostgresHook https://github.com/astronomer/astro-sdk/issues/1291

Reviewer 2: This need to be included in 1.2.3 as per latest discussion.

Author: [Done](https://github.com/astronomer/astro-sdk/pull/1289/commits/2370dd1c2550667d3a2a180268076f59e800061b)

### Improvement:

> * Use `DefaultExtractor` from OpenLineage. `OPENLINEAGE_EXTRACTORS` is not required to be set by user to use OpenLineage. [#1223](https://github.com/astronomer/astro-sdk/issues/1223)

Reviewer 2: can you think from a user's perspective, and rephrase this sentence?

Reviewer 3: I also think this should be part of 1.3 instead of 1.2.3

Author: Sure. We can update documentation accordingly as well for this: https://github.com/astronomer/astro-sdk/issues/1292

Author: Added this to 1.3.0 milestone

### Docs
* Include description on pickling and usage of custom Xcom backend in README.md [#1203](https://github.com/astronomer/astro-sdk/issues/1203)

### Misc

> * Investigate and fix the tests that is filling up Snowflake instances with tables as part of CI. [#738](https://github.com/astronomer/astro-sdk/issues/738)

Reviewer 2: ```suggestion
* Investigate and fix tests that are filling up Snowflake database with tmp tables as part of our CI execution. [#738](https://github.com/astronomer/astro-sdk/issues/738)
```

Author: [Done](https://github.com/astronomer/astro-sdk/pull/1289/commits/7599ef157ac85a6992512c9367fb4368ce9f4e86)

> ### Bug fixes

Author: [Done](https://github.com/astronomer/astro-sdk/pull/1289/commits/7599ef157ac85a6992512c9367fb4368ce9f4e86)
